sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,934 @@
+  async listRules(
+    topicName: string,
+    subscriptionName: string,
+    listRequestOptions?: ListRequestOptions
+  ): Promise<ListRulesResponse> {
+    log.httpAtomXml(
+      `Performing management operation - listRules() with options: ${listRequestOptions}`
+    );
+    const fullPath = this.getSubscriptionPath(topicName, subscriptionName) + "/Rules/";
+    const response: HttpOperationResponse = await this.listResources(
+      fullPath,
+      listRequestOptions,
+      this.ruleResourceSerializer
+    );
+
+    return this.buildListRulesResponse(response);
+  }
+
+  /**
+   * Updates properties on the Rule by the given name based on the given options
+   * @param topicName
+   * @param subscriptionName
+   * @param ruleName
+   * @param ruleOptions Options to configure the Rule being updated.
+   * For example, you can configure the filter to apply on associated Topic/Subscription
+   */
+  async updateRule(
+    topicName: string,
+    subscriptionName: string,
+    ruleName: string,
+    ruleOptions: RuleOptions
+  ): Promise<UpdateRuleResponse> {
+    log.httpAtomXml(
+      `Performing management operation - updateRule() for "${ruleName}" with options: ${ruleOptions}`
+    );
+    const fullPath = this.getRulePath(topicName, subscriptionName, ruleName);
+    const response: HttpOperationResponse = await this.putResource(
+      fullPath,
+      buildRuleOptions(ruleName, ruleOptions),
+      this.ruleResourceSerializer,
+      true
+    );
+
+    return this.buildRuleResponse(response);
+  }
+
+  /**
+   * Deletes a rule.
+   * @param topicName
+   * @param subscriptionName
+   * @param ruleName
+   */
+  async deleteRule(
+    topicName: string,
+    subscriptionName: string,
+    ruleName: string
+  ): Promise<DeleteRuleResponse> {
+    log.httpAtomXml(`Performing management operation - deleteRule() for "${ruleName}"`);
+    const fullPath = this.getRulePath(topicName, subscriptionName, ruleName);
+    const response: HttpOperationResponse = await this.deleteResource(
+      fullPath,
+      this.ruleResourceSerializer
+    );
+
+    return this.buildRuleResponse(response);
+  }
+
+  /**
+   * Creates or updates a resource based on `isUpdate` parameter.
+   * @param name
+   * @param entityFields
+   * @param isUpdate
+   * @param serializer
+   */
+  private async putResource(
+    name: string,
+    entityFields:
+      | InternalQueueOptions
+      | InternalTopicOptions
+      | InternalSubscriptionOptions
+      | InternalRuleOptions,
+    serializer: AtomXmlSerializer,
+    isUpdate: boolean = false
+  ): Promise<HttpOperationResponse> {
+    const webResource: WebResource = new WebResource(this.getUrl(name), "PUT");
+    webResource.body = JSON.stringify(entityFields);
+    if (isUpdate) {
+      webResource.headers.set("If-Match", "*");
+    }
+    webResource.headers.set("content-type", "application/atom+xml;type=entry;charset=utf-8");
+    webResource.headers.set("content-length", Buffer.byteLength(webResource.body));
```

Comment:
Yeah, updated it to use the right value.
In this case, per spec https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.4 it should be passed and value must match for all of the requests since they aren't streaming ones.